src/animation.cpp

```diff
@@ -0,0 +1,429 @@
+#include "game.h"
+#include "debug.h"
+#include "cata_tiles.h" // all animation functions will be pushed out to a cata_tiles function in some manner
+
+extern cata_tiles *tilecontext; // obtained from sdltiles.cpp
+extern void try_update();
+
+// see game.cpp
+bool is_valid_in_w_terrain(int x, int y);
+
+namespace {
+void draw_animation_delay(long const scale = 1)
+{
+    auto const delay = static_cast<long>(OPTIONS["ANIMATION_DELAY"]) * scale * 1000000l;
+
+    timespec const ts = {0, delay};
+    if (ts.tv_nsec > 0) {
+        nanosleep(&ts, nullptr);
+    }
+}
+
+void draw_explosion_curses(WINDOW *const w, int const x, int const y,
+    int const r, nc_color const col)
+{
+    if (r == 0) { // TODO why not always print '*'?
+        mvwputch(w, x, y, col, '*');
+    }
+
+    for (int i = 1; i <= r; ++i) {
```

Comment:
It is (should be?) needed -- this was an oversight on my part when shifting stuff around, and I didn't notice the change when testing.